assets/js/base/components/text-input/validated-text-input.tsx

```diff
@@ -96,34 +96,30 @@ const ValidatedTextInput = ( {
 	);
 
 	/**
-	 * Runs validation on change if the current element is not in focus. This is because autofilled elements do not
-	 * trigger the blur() event.
+	 * Focus on mount
+	 *
+	 * If the input is in pristine state, focus the element.
 	 */
-	const maybeValidateOnChange = useCallback( () => {
-		if (
-			inputRef.current?.ownerDocument.activeElement !== inputRef.current
-		) {
-			validateInput( true );
-		}
-	}, [ validateInput ] );
-
 	useEffect( () => {
-		if ( isPristine ) {
-			if ( focusOnMount ) {
-				inputRef.current?.focus();
-			}
-			setIsPristine( false );
+		if ( isPristine && focusOnMount ) {
+			inputRef.current?.focus();
 		}
+		setIsPristine( false );
 	}, [ focusOnMount, isPristine, setIsPristine ] );
 
+	/**
+	 * Value Validation
+	 *
+	 * Runs validation on state change if the current element is not in focus. This is because autofilled elements do not
+	 * trigger the blur() event, and so values can be validated in the background if the state changes elsewhere.
+	 */
 	useEffect( () => {
-		if ( isPristine ) {
-			if ( validateOnMount ) {
-				validateInput();
-			}
-			setIsPristine( false );
+		if (
+			inputRef.current?.ownerDocument?.activeElement !== inputRef.current
+		) {
+			validateInput( true );
 		}
-	}, [ isPristine, setIsPristine, validateOnMount, validateInput ] );
+	}, [ value, validateInput ] );
```

Comment:
This might get removed by mistake given it's not used in the use effect, a comment would help protect against that.
```suggestion
		// We need to track value even if it's directly used.
	}, [ value, validateInput ] );
```